docassemble/ALWeaver/custom_values.py

```diff
@@ -178,6 +180,63 @@ def get_yml_deps_from_choices(choices: Union[List[str], DADict]):
         return choices
 
 
+def get_full_dep_details(dep_category=None):
+    """Get the full dictionary describing each dependency (including custom ones) so that it can be
+    filtered and used as needed."""
+    dep_choices = []
+
+    # TODO: do we want to prefix the choice with the package name?
+    for capability in _al_weaver_capabilities:
+        if dep_category == "organization":
+            dep_choices.extend(
+                _al_weaver_capabilities[capability].get("organization_choices", [])
+            )
+        elif dep_category == "jurisdiction":
+            dep_choices.extend(
+                _al_weaver_capabilities[capability].get("jurisdiction_choices", [])
+            )
+
+    return list(unique_everseen(dep_choices))
+
+
+def get_matching_deps(dep_category=None, state=None):
+    """Get the dependencies that match the specified state.
+    State "ANY" has a special meaning."""
+    dep_choices = []
+
+    # TODO: do we want to prefix the choice with the package name?
+    for capability in _al_weaver_capabilities:
+        if dep_category == "organization":
+            dep_choices.extend(
+                [
+                    item.get("include_name")
+                    for item in _al_weaver_capabilities[capability].get(
+                        "organization_choices", []
+                    )
+                    if item.get("state", "").lower() == state.lower()
```

Comment:
I believe this will fail when we have the default of `state=None` (which can't happen in the interview, but could if we add other calls to this function): what's supposed to be the difference in behavior between None vs Any?

Reply:
Good catch on the type mismatch.

`None` means that the author hasn't specified that this dependency is usable for any specific state (i.e., it will never be selected automatically). `ANY` means that the dependency is explicitly relevant for *all* states, and will be automatically selected if there's no more specific dependencys that match the state. Right now the `ALAnyState` repo is the only dependency that has state agnostic branding. But if we added a "CourtFormsOnline" package and decided to branch out into multiple states, we wouldn't want to always apply that brand to every interview that uses the "I'm feeling lucky" button.